--- origsrc/mail-notification-5.4/jbsrc/jb.c	2008-05-22 10:47:04.000000000 -0500
+++ src/mail-notification-5.4/jbsrc/jb.c	2008-05-23 14:08:33.906250000 -0500
@@ -26,6 +26,12 @@
 #jb_include <jb-sasl2>
 #jb_include <jb-string-arch-unaligned>
 
+#if defined(__CYGWIN__)
+#define EXEEXT ".exe"
+#else
+#define EXEEXT
+#endif
+
 #define MN_WARNING_CFLAGS \
   "-Wall "				\
   "-Wformat-y2k "			\
@@ -411,7 +420,7 @@
 					"GETTEXT_PACKAGE", "$package",
 					NULL);
 
-  object = JB_OBJECT(jb_program_new("mail-notification"));
+  object = JB_OBJECT(jb_program_new("mail-notification" EXEEXT));
 
   if (jb_variable_get_bool("glibc"))
     {
